### PR TITLE
Features/#36 update airflow

### LINKED ISSUE #36: Update Apache Airflow version
The current apache airflow version is outdated. We want to update it to avoid further requirement issues when adding new packages. That shall lead into updating other packages. Some we know already are:

- pandas
- sqlalchemy

Then (since a newer pandas version is needed), we can include the tasks which download foreign demand and generation data from entsoe via their API using the package entsoe-py (see [here](https://github.com/openego/powerd-data/blob/097a97bfc1b7debc1ab4948ca524b71787ebb77e/src/egon/data/datasets/electrical_neighbours.py#L1285-L1361))

### PR DESCRIPTION
**Pull request by ClaraBuettner**
Fixes #36  .

### CODE REVIEW

File: src/egon/data/datasets/scenario_capacities.py [hunk 1]
```diff
@@ -813,7 +817,11 @@ def eGon100_capacities():
...
 
-if "eGon2035" in egon.data.config.settings()["egon-data"]["--scenarios"]:
+if (
+    "eGon2035" in egon.data.config.settings()["egon-data"]["--scenarios"]
+) and not (
+    "status2019" in egon.data.config.settings()["egon-data"]["--scenarios"]
+):
     tasks = tasks + (insert_data_nep,)
 
 if "eGon100RE" in egon.data.config.settings()["egon-data"]["--scenarios"]:
```
Comment (CarlosEpia) [inline src/egon/data/datasets/scenario_capacities.py:827]: Does it mean that for eGon100RE the task "insert_data_nep" should not be executed?

**Review by CarlosEpia — COMMENTED** [review]
During the installation in a new virtual environment, there were problems with the package psycopg2. Upgrading setuptools from  59.6.0 to 67.8.0 solved the problem. After this, the installation was successful.

When executing egon-data serve, the following problem was found:
`AttributeError: module 'importlib_resources' has no attribute 'read_text'`
This was solved by downgrading importlib-sources from 6.0.1 to 5.12.0.

Could be worth including a comment about the package docker-compose in the installation instructions: It must be installed using pip when it is not installed globally.

**Comment by ClaraBuettner** [comment]
The problem with the username and password when opening the airflow web interface should be fixed now.

**Review by CarlosEpia — APPROVED** [review]
Once the comments about dependencies are addressed, this PR can be merged.

**Comment by ClaraBuettner** [comment]
Adding `docker-compose` to the install-requires resulted in problems with `cdsapi`. That is why I did not include it here.

**Pull request merged**